tests/devices/i19/access_controlled/test_attenuator_motor_squad.py

```
@@ -0,0 +1,128 @@
+    expected_request: dict = {
+        "name": expected_request_name,
+        "params": expected_parameters,
+        "instrument_session": expected_instrument_session,
+    }
+    internal_setter.assert_called_once_with(expected_request)
+
+
+@pytest.mark.parametrize("invoking_hutch", [HutchState.EH1, HutchState.EH2])
+@patch("dodal.devices.i19.access_controlled.blueapi_device.ClientSession.post")
+async def test_when_rest_post_unsuccessful_that_error_raised(
+    unsuccessful_post, invoking_hutch
+):
+    motors: AttenuatorMotorSquad = await given_a_squad_of_attenuator_motors(
+        invoking_hutch
+    )
+    with pytest.raises(ClientConnectionError):
+        restful_response: AsyncMock = given_an_unhappy_restful_response()
+        unsuccessful_post.return_value.__aenter__.return_value = restful_response
+
+        postion_demands = given_position_demands()
+        await motors.set(postion_demands)
+
+
+@pytest.mark.parametrize("invoking_hutch", [HutchState.EH1, HutchState.EH2])
+@patch("dodal.devices.i19.access_controlled.blueapi_device.LOGGER")
+@patch("dodal.devices.i19.access_controlled.blueapi_device.ClientSession.put")
+@patch("dodal.devices.i19.access_controlled.blueapi_device.ClientSession.post")
+async def test_that_error_is_logged_when_response_to_position_demand_set_indicates_failure(
+    restful_post, restful_put, logger, invoking_hutch
+):
+    response_to_post: AsyncMock = AsyncMock()
+    response_to_post.ok = True
+    response_to_post.json.return_value = {"task_id": "0123"}
+    restful_post.return_value.__aenter__.return_value = response_to_post
+
+    response_to_put = AsyncMock()
+    response_to_put.ok = False
+    restful_put.return_value.__aenter__.return_value = response_to_put
+
+    motors: AttenuatorMotorSquad = await given_a_squad_of_attenuator_motors(
+        invoking_hutch
+    )
+    position_demands: AttenuatorMotorPositionDemands = given_position_demands()
+
+    logger.error.assert_not_called()
```

The await should raise an exception instead of succeeding.